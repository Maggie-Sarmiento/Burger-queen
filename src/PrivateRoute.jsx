/* eslint-disable react/prop-types */
import React from 'react';
import { Routes, Route } from 'react-router-dom';
import LogIn from './Components/No-Authenticate/Login/LogIn';
import Staff from './Components/Authenticate/Admin/Staff/Staff';
import Menu from './Components/Authenticate/Admin/Menu/Menu';
import MenuLunch from './Components/Authenticate/Admin/LunchMenu/MenuLunch';
import Home from './Components/Authenticate/Common/Home';

const PrivateRoute = ({ role, authenticate }) => {
  if (role === 'Admin') {
    return (
      <Routes>
        <Route path="/" element={<LogIn />} />
        <Route path="/home/*" element={<Home role={role} authenticate={authenticate} />}>
          <Route path="staff" element={<Staff />} />
          <Route path="menu" element={<Menu role={role} />} />
          <Route path="menu-lunch" element={<MenuLunch role={role} />} />
        </Route>
      </Routes>
    );
  }

  if (role === 'Mesero') {
    return (
      <Routes>
        <Route path="/" element={<LogIn />} />
        <Route path="/home/*" element={<Home role={role} authenticate={authenticate} />}>
          <Route path="menu" element={<Menu role={role} />} />
          <Route path="menu-lunch" element={<MenuLunch role={role} />} />
          <Route path="order" element={<Menu role={role} authenticate={authenticate} />} />
          <Route path="toserve" element={<Menu role={role} authenticate={authenticate} />} />
        </Route>
      </Routes>
    );
  }

  if (role === 'Jefe de cocina') {
    return (
      <Routes>
        <Route path="/" element={<LogIn />} />
        <Route path="/home/*" element={<Home role={role} authenticate={authenticate} />}>
          <Route path="order" element={<Menu role={role} authenticate={authenticate} />} />
          <Route path="toserve" element={<Menu role={role} authenticate={authenticate} />} />
        </Route>
      </Routes>
    );
  }

  return null;
};

export default PrivateRoute;
