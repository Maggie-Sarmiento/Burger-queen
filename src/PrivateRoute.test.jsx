/* eslint-disable react/jsx-curly-brace-presence */
/* eslint-disable jsx-a11y/aria-role */
/* eslint-disable no-undef */
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { create, act } from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import PrivateRoute from './PrivateRoute';

describe('test PrivateRouter', () => {
  const mockAuthenticate = jest.fn();
  test('render PrivateRouter', async () => {
    let root;
    act(() => {
      root = create(
        <Router>
          <PrivateRoute authenticate={mockAuthenticate} />
        </Router>,
      );
    });
    expect(root.toJSON()).toMatchSnapshot();
  });
  test('render conditional PrivateRouter', async () => {
    let value = 'Admin';
    const { rerender } = render(
      <Router>
        <PrivateRoute role={value} authenticate={mockAuthenticate} />
      </Router>,
    );
    expect(screen.getByText(/Inicia sesión para administar el staff y menús o para tomar la orden de los clientes./i)).toBeInTheDocument();
    value = 'Mesero';
    rerender(
      <Router>
        <PrivateRoute role={value} authenticate={mockAuthenticate} />
      </Router>,
    );
    value = 'Jefe de cocina';
    rerender(
      <Router>
        <PrivateRoute role={value} authenticate={mockAuthenticate} />
      </Router>,
    );
  });
});
