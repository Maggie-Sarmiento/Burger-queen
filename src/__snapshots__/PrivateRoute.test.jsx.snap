// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test PrivateRouter render PrivateRouter 1`] = `null`;
